src/metaprob/state.clj

```diff
@@ -89,8 +95,7 @@
 
 ;; Constructors
 
-(defn empty-state []
-  '())
+(defn empty-state [] {})    ; 'lein test' passes with () and [] here as well
```

Comment:
If we're committed to the immutable path, and I don't know what the state of that discussion is, then we could use `def` instead of `defn` here.

Reply:
I don't understand?

Reply:
If we're planning on removing support for mutable traces `empty-state` could be the empty state value itself rather than a function that returns the empty state value:
```
(def empty-state {})
```
Does that make sense?